lib/livereload.coffee

```diff
@@ -69,18 +69,32 @@ class Server
     @debug "Browser disconnected."
 
   watch: (paths) ->
-    @watcher = chokidar.watch paths, {ignoreInitial: true, ignored: @config.exclusions, usePolling: @config.usePolling}
-    @watcher.on 'add', (path) => @filterRefresh path
-    @watcher.on 'change', (path) => @filterRefresh path
-    @watcher.on 'unlink', (path) => @filterRefresh path
+    @watcher = chokidar.watch(paths,
+      ignoreInitial: true
+      ignored: @config.exclusions
+      usePolling: @config.usePolling
+    )
+    .on 'add', @filterRefresh.bind(@)
+    .on 'change', @filterRefresh.bind(@)
+    .on 'unlink', @filterRefresh.bind(@)
+
 
   filterRefresh: (filepath) ->
     exts = @config.exts
     fileext = path.extname filepath
                   .substring 1
-    for ext in exts when ext == fileext
-      @refresh filepath
-      break
+
+    # check if file extension is supposed to be watched
+    if (exts.indexOf(fileext) != -1)
```

Comment:
Personally I think this syntax is clearer to the purpose of the logic, which is to determine whether the extension is included in the developer's watched extensions and call `.refresh` once and only once.